### PR TITLE
Add JSONAdaptedAppointment

### LINKED ISSUE #84: Add storage for appointments

### PR DESCRIPTION
**Pull request by chiangqinkang**
Resolves #84

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (time == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()));
+        }
+        if (!Time.isValidTime(time)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
+        }
+        final Time modelTime = new Time(time);
+
+        return new Appointment(modelId, modelDoctor, modelPatient, modelDate, modelTime);
```
Comment (MiloTruck) [inline src/main/java/seedu/address/storage/JsonAdaptedAppointment.java:100]: I think @sam-theman88 is removing the following two lines from the constructor of `Appointment` in his PR (see [this commit](https://github.com/AY2425S1-CS2103T-T09-3/tp/pull/89/commits/eb411b511f8cf0548a2cc2048aa740282e3d45f8)):

```java
        doctor.addAppointment(this);
        patient.addAppointment(this);
```

Do you need to call `addAppointment()` manually here if these two lines are removed?

**Review by taniathng — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -94,6 +98,23 @@ public void removePerson(Person key) {
...
+     */
+    public boolean hasAppointment(Appointment appointment) {
+        requireNonNull(appointment);
+        return appointments.contains(appointment);
+    }
+
+    /**
+     * Adds a person to the address book.
+     * The person must not already exist in the address book.
+     */
```
Comment (bmanara) [inline src/main/java/seedu/address/model/AddressBook.java:113]: I think forgot to change comment? I'll change this once we merge

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java (hunk 1 above)
Comment (bmanara) [reply src/main/java/seedu/address/storage/JsonAdaptedAppointment.java:100]: Discussed with @sam-theman88, we will add back the two lines and handle duplicates in the constructor of `Appointment` itself.

**Review by bmanara — APPROVED** [review]
LGTM

**Pull request merged**